lib/compat/wordpress-5.9/class-wp-theme-json-gutenberg.php

```diff
@@ -1168,7 +1168,7 @@ private static function flatten_tree( $tree, $prefix = '', $token = '--' ) {
 			$new_key = $prefix . str_replace(
 				'/',
 				'-',
-				strtolower( preg_replace( '/(?<!^)[A-Z]/', '-$0', $property ) ) // CamelCase to kebab-case.
+				strtolower( _wp_to_kebab_case( $property ) )
```

Comment:
One remark, `_wp_to_kebab_case` function was added in WordPress 5.8. Have you tested it with WP 5.7?

Reply:
Yeah, we use that function in many places of the codebase and it's [defined by the plugin](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-5.8/utils.php#L79) for 5.7.

Reply:
👌